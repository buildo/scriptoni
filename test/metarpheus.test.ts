import * as path from 'path';
import { runCommands, testAppDir } from './utils';
import * as fs from 'fs';
import * as rimraf from 'rimraf';

const config = require(path.resolve(testAppDir, 'metarpheus.config.test'));

const metarpheusDir = config.apiOut
  .split('/')
  .slice(0, -1)
  .join('/');
const metarpheusOutPath = path.resolve(testAppDir, metarpheusDir);

afterAll(() => {
  rimraf.sync(metarpheusOutPath);
});

fdescribe('metarpheus', () => {
  describe('metarpheusConfig option', () => {
    it('should read the correct config file', () => {
      const apiPath = path.resolve(testAppDir, config.apiOut);
      const modelsPath = path.resolve(testAppDir, config.modelOut);

      return runCommands([
        `cd ${testAppDir}`,
        './node_modules/.bin/scriptoni metarpheus --metarpheusConfig metarpheus.config.test.js'
      ]).then(() => {
        expect(fs.existsSync(apiPath)).toBeTruthy();
        expect(fs.existsSync(modelsPath)).toBeTruthy();

        expect(fs.readFileSync(apiPath, { encoding: 'utf8' })).toMatchSnapshot();
        expect(fs.readFileSync(modelsPath, { encoding: 'utf8' })).toMatchSnapshot();
      }, err => {
        console.log((err as any).message) return Promise.reject(err)
      });
    });
  });
});
