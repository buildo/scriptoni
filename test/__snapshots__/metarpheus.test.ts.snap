// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`metarpheus metarpheusConfig option should read the correct config file 1`] = `
"API_PRELUDE

export default function getRoutes(_metarpheusRouteConfig: RouteConfig) {
  return {

  }
}"
`;

exports[`metarpheus metarpheusConfig option should read the correct config file 2`] = `
"// DO NOT EDIT MANUALLY - metarpheus-generated
import * as t from 'io-ts'
// @ts-ignore
import { createOptionFromNullable } from 'io-ts-types/lib/fp-ts/createOptionFromNullable'
// @ts-ignore
import { Option } from 'fp-ts/lib/Option'

MODEL_PRELUDE
export const VoidFromUnit = new t.Type<void, {}>(
  'VoidFromUnit',
  (_m): _m is void => true,
  () => t.success(undefined),
  () => ({})
)
export interface Account {
  id: string,
  username: string,
  email: Option<string>,
  publicKey: string
}

export const Account = t.type({
  id: t.string,
  username: t.string,
  email: createOptionFromNullable(t.string),
  publicKey: t.string
}, 'Account')"
`;
